File: employee-management-system-ui/src/service/EmployeeService.js
import axios from "axios";

const API_BASE_URL = "http://localhost:8080/api/v1/employee";


class EmployeeService {
   addEmployee(employee){
       return axios.post(API_BASE_URL,employee);     
   }
   getAllEmployees(){
     return axios.get(API_BASE_URL);
   }
   deleteEmployee(id){
    return axios.delete(`${API_BASE_URL}/${id}`)
   }
   getEmployeeById(id){
    return axios.get(`${API_BASE_URL}/${id}`);
   }
   updateEmloyee(id,employee){
    return axios.put(`${API_BASE_URL}/${id}`,employee);
   }
}

export default new EmployeeService();

File: employee-management-system-api/src/main/java/com/formation/api/service/EmployeeService.java
package com.formation.api.service;

import com.formation.api.model.Employee;
import com.formation.api.repository.EmployeeRepository;

import java.util.List;

public interface EmployeeService {
    Employee createEmployee(Employee employee);

    List<Employee> getAllEmployees();

    Boolean deleteEmployee(Long id);

    Employee getEmployeeById(Long id);

    Employee updateEmployee(Long id, Employee employee);
}
